dtfs-central-api/src/helpers/get-fee-record-payment-entity-group-status.test.ts

```diff
@@ -0,0 +1,70 @@
+
+describe('getFeeRecordPaymentEntityGroupStatus', () => {
+  it('throws an error when the group has an empty list of fee records', () => {
+    // Arrange
+    const group: FeeRecordPaymentEntityGroup = {
+      feeRecords: [],
+      payments: [],
+    };
+
+    // Act / Assert
+    expect(() => getFeeRecordPaymentEntityGroupStatus(group)).toThrow(new Error('Fee record payment entity group cannot have an empty fee records array'));
+  });
+
+  it.each(Object.values(FEE_RECORD_STATUS))('returns the group status when each fee record in the group has status %s', (status) => {
+    // Arrange
+    const feeRecords = [
+      FeeRecordEntityMockBuilder.forReport(utilisationReport()).withStatus(status).build(),
+      FeeRecordEntityMockBuilder.forReport(utilisationReport()).withStatus(status).build(),
+      FeeRecordEntityMockBuilder.forReport(utilisationReport()).withStatus(status).build(),
+    ];
+    const group: FeeRecordPaymentEntityGroup = { feeRecords, payments: [] };
+
+    // Act
+    const result = getFeeRecordPaymentEntityGroupStatus(group);
+
+    // Assert
+    expect(result).toEqual(status);
+  });
+
+  it(`returns the status ${FEE_RECORD_STATUS.READY_TO_KEY} when the fee records in the group have both the ${FEE_RECORD_STATUS.READY_TO_KEY} and ${FEE_RECORD_STATUS.RECONCILED} status`, () => {
+    // Arrange
+    const feeRecords = [
+      FeeRecordEntityMockBuilder.forReport(utilisationReport()).withStatus(FEE_RECORD_STATUS.READY_TO_KEY).build(),
+      FeeRecordEntityMockBuilder.forReport(utilisationReport()).withStatus(FEE_RECORD_STATUS.READY_TO_KEY).build(),
+      FeeRecordEntityMockBuilder.forReport(utilisationReport()).withStatus(FEE_RECORD_STATUS.RECONCILED).build(),
+    ];
+    const group: FeeRecordPaymentEntityGroup = { feeRecords, payments: [] };
+
+    // Act
+    const result = getFeeRecordPaymentEntityGroupStatus(group);
+
+    // Assert
+    expect(result).toEqual(FEE_RECORD_STATUS.READY_TO_KEY);
+  });
+
+  it.each([FEE_RECORD_STATUS.TO_DO, FEE_RECORD_STATUS.DOES_NOT_MATCH, FEE_RECORD_STATUS.MATCH])(
+    'throws an error when all the fee records in the group have status %s expect for one',
+    (status) => {
+      // Arrange
+      const feeRecords = [
+        FeeRecordEntityMockBuilder.forReport(utilisationReport()).withStatus(status).build(),
+        FeeRecordEntityMockBuilder.forReport(utilisationReport()).withStatus(status).build(),
+        FeeRecordEntityMockBuilder.forReport(utilisationReport()).withStatus(FEE_RECORD_STATUS.READY_TO_KEY).build(),
+      ];
+      const group: FeeRecordPaymentEntityGroup = { feeRecords, payments: [] };
+
+      // Act / Assert
+      expect(() => getFeeRecordPaymentEntityGroupStatus(group)).toThrow(
+        new Error(`Fee record payment entity group has an invalid set of statuses: [${status}, ${FEE_RECORD_STATUS.READY_TO_KEY}]`),
+      );
+    },
+  );
+
+  function utilisationReport(): UtilisationReportEntity {
```

Comment:
This function is now duplicated in 14 places, we should create a DRY test helper @Zainzzkk

Reply:
Separate PR raised here: https://github.com/UK-Export-Finance/dtfs2/pull/3465 Will update whichever PR goes in second to use the dry helper here also.